staging/src/k8s.io/apiserver/pkg/admission/plugin/validatingadmissionpolicy/typechecking.go

```diff
@@ -47,124 +47,162 @@ type TypeChecker struct {
 	restMapper     meta.RESTMapper
 }
 
+// TypeCheckingContext holds information about the policy being type-checked.
+// The struct is opaque to the caller.
+type TypeCheckingContext struct {
+	gvks          []schema.GroupVersionKind
+	declTypes     []*apiservercel.DeclType
+	paramGVK      schema.GroupVersionKind
+	paramDeclType *apiservercel.DeclType
+}
+
 type typeOverwrite struct {
 	object *apiservercel.DeclType
 	params *apiservercel.DeclType
 }
 
-// typeCheckingResult holds the issues found during type checking, any returned
+// TypeCheckingResult holds the issues found during type checking, any returned
 // error, and the gvk that the type checking is performed against.
-type typeCheckingResult struct {
+type TypeCheckingResult struct {
 	gvk schema.GroupVersionKind
 
 	issues *cel.Issues
 	err    error
 }
 
+// GVK returns the GroupVersionKind associate with the result.
+func (r *TypeCheckingResult) GVK() schema.GroupVersionKind {
+	return r.gvk
+}
```

Comment:
Consider either exporting `gvks` field (if callers can be trusted to not modify the slice), or copying the slice here (if callers cannot be trusted)?

Reply:
Let's export these. The struct is a result type. The caller has the ownership of it.

Reply:
Done. PTAL. Thank you.

Reply:
Diff: https://github.com/kubernetes/kubernetes/compare/26c25744dfbbc52f8746db01b41b608a8d7fadd5..ee2bb1b1b27cb2232c8355ff172b6966b25c4c75